commit e2696df9e25ac9ce03b122a2973d5b008ebcaf62
Author: lesssalt <lesssalt@proton.me>
Date:   2024-10-02T00:10:02+00:00

    Fix selection of nop menu items
    
    Nop menu items are normally not selectable, but this is not checked when
    the menu is initially spawned. We iterate through the items until we
    find an actionable item to select and highlight.

---
 menu.c |   13 ++++++++++---
 1 file changed, 10 insertions(+), 3 deletions(-)

--- a/menu.c
+++ b/menu.c
@@ -479,9 +479,11 @@ select_menu(Menu *menu, int mselect)
 	menu->select < menu->offset + menu->height)
 	draw_menu_item(menu, menu->select);
     menu->select = mselect;
-    standout();
-    draw_menu_item(menu, menu->select);
-    standend();
+    if (menu->item[menu->select].type != MENU_NOP) {
+	standout();
+	draw_menu_item(menu, menu->select);
+	standend();
+    }
     move(menu->y + mselect - menu->offset, menu->x);
     toggle_stand();
     refresh();
@@ -615,6 +617,11 @@ popup_menu(Menu *parent, Menu *menu)
 	menu->cursorY = parent->cursorY;
 	guess_menu_xy(parent, menu->width, &menu->x, &menu->y);
     }
+    while (menu->item[menu->select].type == MENU_NOP) {
+	if (menu->select >= menu->nitem - 1)
+	    break;
+	menu->select++;
+    }
     geom_menu(menu, menu->select);
 
     CurrentMenu = menu;
